rust/crd/src/lib.rs

```diff
@@ -365,6 +382,21 @@ fn env_var_from_secret(var_name: &str, secret: &str, secret_key: &str) -> EnvVar
     }
 }
 
+fn env_var_from_config_map(var_name: &str, config_map: &str, config_map_key: &str) -> EnvVar {
```

Comment:
We already added something like this recently: https://github.com/stackabletech/operator-rs/pull/370
But it's a function on the ContainerBuilder. And in the end I decided against using the functions like this because it would make the code worse. And Sebastian and I didn't want two ways to do the same thing to keep the operators more "the same". In the end I decided to use the function here with the thought "this operator is different from all the others"